src/client.cpp

```diff
@@ -413,8 +413,9 @@ Rectangle HSClient::outer_floating_rect() {
     return dec.inner_to_outer(float_size_);
 }
 
-int close_command(int argc, char** argv, Output) {
-    auto window = get_window((argc > 1) ? argv[1] : "");
+int close_command(Input input, Output) {
+    std::string winid;
+    auto window = get_window((input >> winid) ? winid : "");
```

Comment:
Is this better than `get_window((input.empty() ? "" : input.front())`? I personally don't like the shift being encapsulated like that (side-effect hidden in forming a method call).

Reply:
Yes I don't see a point in this kind of consistency, I would use the accessor that fits best in the circumstances. For example, there is other places where the number of arguments is unknown, and there it's best to just have a for-loop.

Reply:
I agree with @ypnos' concerns, but I also think that we kind of paved the way for non-ideal situations like this when we introduced `>>` such that it has a meaningful return value.

Elsewhere, this mostly just leads to `if (input >> foo) {…}`, which is only slightly less bad than wrapping a function call around it like here, in my opinion.

So while I'm with @thorsten here in that I'd prefer sticking to `>>` whenever possible, I see the point of not sacrificing readability for consistency.

So maybe there is a way to use `>>` but retain readability. Wouldn't it be possible to simply put it like this?

```
std::string winid;
input >> winid;
auto window = get_window(winid);
```

Reply:
Yes this is possible. The usage within `if` for me is already bad style (side-effect of an if-condition! noes!) but it is tolerable because it is somewhat contained and makes the code much more readable where we use it. (Also look at http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2016/p0305r0.html urgs).

I do not see the rationale behind using >> whenever possible. It should be used where it works best. Writing three lines instead of one that is simply understood for me is not best use.

Reply:
Well, from my pov, it's not about using `>>` whenever possible, but about providing an interface that is universally preferred. I had hoped that the stream operator is able to do that, and I view every situation where we "fall back" to `input.front()` as reason to reconsider the merits of `>>`. Maybe it can still be improved?

That being said, it is of course always possible that the best solution is two solutions. That seems to be the case with ArgList's interfaces.

The points you make against my suggestion are perfectly valid, nothing to argue there.

Reply:
I've updated the function to
```
int close_command(Input input, Output) {
    std::string winid = "";
    input >> winid; // try to read, use "" otherwise
    auto window = get_window(winid);
    if (window != 0)
        window_close(window);
    else return HERBST_INVALID_ARGUMENT;
    return 0;
}
```
If you want to discuss the usage of `>>` please open a issue, but it's not in the scope of this PR.